iptables/iptables.go

```diff
@@ -241,6 +241,26 @@ func (ipt *IPTables) DeleteIfExists(table, chain string, rulespec ...string) err
 	return err
 }
 
+// DeleteById deletes the rule with the specified ID in the given table and chain.
+func (ipt *IPTables) DeleteById(table, chain string, id int) (string, error) {
+	rules, err := ipt.List(table, chain)
+	if err != nil {
+		return "", err
+	}
+
+	for _, rule := range rules {
+		if strings.HasPrefix(rule, strconv.Itoa(id)+" ") {
```

Comment:
Is this check really necessary? If id is the index in the chain, then why not just unconditionally delete it?

Reply:
Oh yes that makes sense, Ill make sure to drop the check in the next commit
Thanks for the review!